sled-agent/src/illumos/dladm.rs

```diff
@@ -164,16 +170,31 @@ impl Dladm {
         Ok(())
     }
 
-    /// Returns all VNICs that may be managed by the Sled Agent.
-    pub fn get_vnics() -> Result<Vec<String>, GetVnicError> {
+    /// Returns VNICs that may be managed by the Sled Agent of a particular
+    /// kind.
+    pub fn get_vnics(kind: VnicKind) -> Result<Vec<String>, GetVnicError> {
         let mut command = std::process::Command::new(PFEXEC);
         let cmd = command.args(&[DLADM, "show-vnic", "-p", "-o", "LINK"]);
         let output = execute(cmd).map_err(|err| GetVnicError { err })?;
 
         let vnics = String::from_utf8_lossy(&output.stdout)
             .lines()
-            .filter(|vnic| vnic.starts_with(VNIC_PREFIX))
-            .map(|s| s.to_owned())
+            .filter_map(|name| {
+                // Ensure this is a kind of VNIC that the sled agent could be
+                // responsible for.
+                match VnicKind::from_name(name) {
+                    Some(vnic_kind) => {
+                        // Ensure matches the caller-specified VnicKind
+                        if vnic_kind == kind {
+                            Some(name.to_owned())
+                        } else {
+                            None
+                        }
+                    }
```

Comment:
Okay, *one* last thing - and sorry for all this churn, I didn't realize we'd shuffle the XDE devices out of the `VnicKind` -

The only time we call `get_vnics`, we do it for *both* kinds, then chain them together. Should we just be returning *all* vnics here that can be managed by the sled agent, and let the caller do the disambiguation?

Basically, remove the arguments to `get_vnics`, and change the impl to the following:

```suggestion
                    Some(vnic_kind) => Some(name.to_owned()),
```

Reply:
No worries at all, there's a lot of tendrils here.

That seems OK to me. We're pretty conservative in what `get_vnics()` will even return, so I don't see any harm in returning everything and letting the caller dispatch on the kind. Will do!

I'll update the one call site as well :)

Reply:
Done in [c3b3329](https://github.com/oxidecomputer/omicron/pull/1052/commits/c3b332931f0a3437bdc454995fe84b9e36ecd273)